civet/compilers/es6.py

```diff
@@ -0,0 +1,57 @@
+import errno
+import os
+
+from civet.compilers.base_compiler import Compiler
+
+
+def mkdir_p(path):
+    try:
+        os.makedirs(path)
+    except OSError as exc:  # Python >2.5
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise
+
+
+class ES6Compiler(Compiler):
+    """Civet compiler for Ecmascript 6 using Babel.
+    """
+    def __init__(self, precompiled_assets_dir, kill_on_error):
+        super(ES6Compiler, self).__init__(precompiled_assets_dir,
+                                          kill_on_error)
+        self.args = [('--compile', '--map')]
+
+    @property
+    def name(self):
+        return "Ecmascript 6"
+
+    @property
+    def executable_name(self):
+        return 'babel'
+
+    @property
+    def executable_setting(self):
+        return 'CIVET_BABEL_BIN'
+
+    def matches(self, base, ext):
+        return ext == '.es6'
```

Comment:
Although I saw the conversation about this in our internal Github, I do think that because this library is open source, we should match conventions and have the ES6 compiler pick up all files ending in `.js`. As a compromise, how about we let this class match on `.js`, but we internally subclass this and override the `matches()` method to match on `.es6`? That way, `civet` can follow standard conventions, but then we're free to break those conventions internally in order to maintain backwards compatibility?